pkg/endpoint/policy.go

```diff
@@ -598,6 +565,50 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 	return needToRegenerateBPF, nil
 }
 
+// updateEndpointOptions updates the boolean configuration options for the endpoint
+// based off of policy configuration, daemon policy enforcement mode, and any
+// configuration options provided in opts. Returns whether the options changed
+// from prior endpoint configuration. Note that the policy which applies
+// to the endpoint, as well as the daemon's policy enforcement, may override
+// configuration changes which were made via the API that were provided in opts.
+// Must be called with endpoint mutex held.
+func (e *Endpoint) updateEndpointOptions(owner Owner, opts models.ConfigurationMap) (optsChanged bool) {
```

Comment:
This function might create some confusion with one that exists already
https://github.com/cilium/cilium/blob/adf4e4f5b31f6d020969aeea1a0a45c6af270cc8/pkg/endpoint/endpoint.go#L1552-L1556

Reply:
Can you please suggest something that you would have instead of `updateEndpointOptions`? What about `overrideEndpointOptions`?

Reply:
@aanm I've made it more explicit, let me know what you think.